### ISSUE TITLE
Add beautiful citation directions via CITATION.cff in template

### ISSUE THREAD

**Issue by d4straub** [nf-core/tools#3360]
### Description of feature

The [CITATION.cff](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files) in a github repo adds a handy citation field and also enhances zenodo entries substantially.

For example, the title and author list of https://zenodo.org/records/14332397 was improved to https://zenodo.org/records/14409147 by the addition of the cff file. That seems a considerable improvement to me.

I do not want to suggest to include in the cff file actual papers, but info about how to cite the software repo. That would also give a good platform to add ORCIDs and all (? guideline incoming?) code contributors.

Possibly the file could be generated partially automatic?

**Comment by muffato** [comment]
Thanks for creating the issue, @d4straub .

I should add a counterpoint to that. We also want to cite all the software used within the pipeline, which are currently manually tracked in `CITATIONS.md`. Text is typically something like

> We ran the pipeline (A et al), which does X (B et al) and Y (C et al)

While CITATION.cff is great for the integration with Zenodo, it may give the wrong impression that all that needs to be cited is the pipeline itself.

**Comment by mirpedrol** [comment]
https://github.com/nf-core/tools/pull/2059
`CITATIONS.cff` had to be removed before because it was causing problems with Zenodo :(

**Comment by d4straub** [comment]
@mirpedrol Yes, but this was because the `CITATIONS.cff` contained the nf-core publication, that would certainly lead to an unwanted result. Quote from above:

> I do not want to suggest to include in the cff file actual papers, but info about how to cite the software repo.

**Comment by muffato** [comment]
> [#2059](https://github.com/nf-core/tools/pull/2059) `CITATIONS.cff` had to be removed before because it was causing problems with Zenodo :(

lol, I can see I approved that one 🙈 ! Like @d4straub said, the issue was it that was the CITATIONS.cff _of nf-core_, meaning that all Zenodo uploads were getting authors, titles, etc, of the nf-core paper. What we'd like here is to get a CITATIONS.cff that truly reflects the pipeline

**Comment by d4straub** [comment]
I thought that it appears in Zendodo at the Notes: section, as below. But I might be mistaken (never tried, no experience, only conjecture).

![Image](https://github.com/user-attachments/assets/b7b3f75b-d497-414b-8127-bb1213574b3f)

**Comment by muffato** [comment]
My bad. Yes, it's there
![Image](https://github.com/user-attachments/assets/17f7f022-d211-4c3b-ba58-5f9ef019222f)
